axelor-sale/src/main/java/com/axelor/apps/sale/service/cartline/CartLineCreateServiceImpl.java

```diff
@@ -57,4 +57,16 @@ public CartLine createCartLine(Cart cart, Product product) throws AxelorExceptio
     cartLine.setCart(cart);
     return cartLineRepository.save(cartLine);
   }
+
+  @Override
+  @Transactional(rollbackOn = Exception.class)
+  public CartLine createCartLineWithVariant(Cart cart, Product product) throws AxelorException {
+    CartLine cartLine = new CartLine();
+    cartLine.setProduct(product.getParentProduct());
+    cartLine.setUnit(saleOrderLineProductService.getSaleUnit(cartLine.getProduct()));
+    cartLine.setVariantProduct(product);
+    cartLine.setPrice(cartLineProductService.getSalePrice(cart, cartLine));
```

Comment:
After settings the variant product, you can use CartLineProductService#getProductInformation to retrieve the informations.